internal/handlers/order.go

```diff
@@ -194,21 +220,82 @@ func (o Order) ListOrderFunc(repository domain.OrderRepository) orders.GetAllOrd
 		if p.Status != nil {
 			_, ok := domain.AllOrderStatuses[*p.Status]
 			if !ok {
-				return orders.NewGetAllOrdersDefault(http.StatusBadRequest).
+				return orders.NewGetUserOrdersDefault(http.StatusBadRequest).
 					WithPayload(buildStringPayload(fmt.Sprintf("Invalid order status '%v'", *p.Status)))
 			}
 		}
 
 		total, err := repository.OrdersTotal(ctx, userID)
 		if err != nil {
 			o.logger.Error("Error while getting total of all user's orders", zap.Error(err))
+			return orders.NewGetUserOrdersDefault(http.StatusInternalServerError).
+				WithPayload(buildErrorPayload(err))
+		}
+
+		var items []*ent.Order
+		if total > 0 {
+			items, err = repository.List(ctx, &userID, orderFilter)
+			if err != nil {
+				o.logger.Error("list items failed", zap.Error(err))
+				return orders.NewGetUserOrdersDefault(http.StatusInternalServerError).WithPayload(buildErrorPayload(err))
+			}
+		}
+
+		mappedOrders, err := mapUserOrdersToResponse(items, o.logger)
+		if err != nil {
+			o.logger.Error("map orders to response failed", zap.Error(err))
+			return orders.NewGetUserOrdersDefault(http.StatusInternalServerError).WithPayload(buildErrorPayload(err))
+		}
+		totalOrders := int64(total)
```

Comment:
I'm not sure that it is a good idea to assign a new variable for the variable `total`. You use `totalOrders` only once in the function below.

Reply:
It's because of conversion of int into *int64. AFAIK this is the only way to do that.